sapi/fpm/fpm/fpm_main.c

```diff
@@ -1157,12 +1157,32 @@ static void init_request_info(void)
 									 * As we can extract PATH_INFO from PATH_TRANSLATED
 									 * it is probably also in SCRIPT_NAME and need to be removed
 									 */
+									char *decoded_path_info = NULL;
+									size_t decoded_path_info_len = 0;
+									if (strchr(path_info, '%')) {
+										decoded_path_info = estrdup(path_info);
+										decoded_path_info_len = php_url_decode(decoded_path_info, strlen(path_info));
+									}
 									int snlen = strlen(env_script_name);
```

Comment:
While you're at it, this is an implicit sign conversion.
```suggestion
									size_t snlen = strlen(env_script_name);
```